wgengine/userspace.go

```diff
@@ -554,6 +560,188 @@ func updateSig(last *string, v interface{}) (changed bool) {
 	return false
 }
 
+// isTrimmablePeer reports whether p is a peer that we can trim out of the
+// network map.
+//
+// We can only trim peers that both a) support discovery (because we
+// know who they are when we receive their data and don't need to rely
+// on wireguard-go figuring it out) and b) for implementation
+// simplicity, have only one IP address (an IPv4 /32), which is the
+// common case for most peers. Subnet router nodes will just always be
+// created in the wireguard-go config.
+func isTrimmablePeer(p *wgcfg.Peer) bool {
+	if len(p.AllowedIPs) != 1 || len(p.Endpoints) != 1 {
+		return false
+	}
+	if !strings.HasSuffix(p.Endpoints[0].Host, ".disco.tailscale") {
+		return false
+	}
+	aip := p.AllowedIPs[0]
+	if aip.Mask != 32 || !aip.IP.Is4() {
```

Comment:
Future proofing for IPv6 support: also allow /128 IPv6?

Or maybe that'll break too much other stuff. In that case, `// TODO: ipv6 support`.

Reply:
```
        // TODO: IPv6 support, once we support IPv6 within the tunnel. In that case,                                                                                                                                                
        // len(p.AllowedIPs) probably will be more than 1.                                                                                                                                                                          
```